services/dask-sidecar/Dockerfile

```diff
@@ -0,0 +1,163 @@
+RUN python -m venv "${VIRTUAL_ENV}"
+RUN pip --no-cache-dir install --upgrade \
+  pip~=21.0.1  \
+  wheel \
+  setuptools
+
+# copy dask-sidecar and dependencies
+COPY --chown=scu:scu packages /build/packages
+COPY --chown=scu:scu services/dask-sidecar /build/services/dask-sidecar
+COPY --chown=scu:scu services/storage/client-sdk /build/services/storage/client-sdk
+COPY --chown=scu:scu services/sidecar /build/services/sidecar
+
+# install base 3rd party dependencies (NOTE: this speeds up devel mode)
+RUN pip --no-cache-dir install \
+  -r /build/services/dask-sidecar/requirements/_base.txt \
+  -r /build/services/dask-sidecar/requirements/_packages.txt
+
+# --------------------------Prod-depends-only stage -------------------
+# This stage is for production only dependencies that get partially wiped out afterwards (final docker image concerns)
+#
+#  + /build
+#    + services/dask-sidecar [scu:scu] WORKDIR
+#
+FROM build as prod-only-deps
+
+WORKDIR /build/services/dask-sidecar
+ENV SC_BUILD_TARGET=prod-only-deps
+RUN pip --no-cache-dir install -r requirements/prod.txt
+
+# --------------------------Production stage -------------------
+# Final cleanup up to reduce image size and startup setup
+# Runs as scu (non-root user)
+#
+#  + /home/scu     $HOME = WORKDIR
+#    + services/dask-sidecar [scu:scu]
+#
+FROM base as production
+
+ENV SC_BUILD_TARGET=production \
+  SC_BOOT_MODE=production
+ENV PYTHONOPTIMIZE=TRUE
+
+WORKDIR /home/scu
+
+# bring installed package without build tools
+COPY --from=prod-only-deps --chown=scu:scu ${VIRTUAL_ENV} ${VIRTUAL_ENV}
+# copy docker entrypoint and boot scripts
+COPY --chown=scu:scu services/dask-sidecar/docker services/dask-sidecar/docker
+
+
+# WARNING: This image is used for dask-scheduler and dask-worker.
+# In order to have the same healty entrypoint port
+# make sure dask worker is started as ``dask-worker --dashboard-address 8787``.
+# Otherwise the worker will take random ports to serve the /health entrypoint.
+HEALTHCHECK \
+  --interval=60s \
+  --timeout=60s \
+  --start-period=10s \
+  --retries=3 \
+  CMD ["curl", "-Lf", "http://127.0.0.1:8787/health"]
+
+ENTRYPOINT [ "/bin/sh", "services/dask-sidecar/docker/entrypoint.sh" ]
+CMD ["/bin/sh", "services/dask-sidecar/docker/boot.sh"]
+
+
+# --------------------------Development stage -------------------
+# Source code accessible in host but runs in container
+# Runs as scu with same gid/uid as host
+# Placed at the end to speed-up the build if images targeting production
+#
+#  + /devel         WORKDIR
+#    + services  (mounted volume)
+#
+FROM build as development
+
+ENV SC_BUILD_TARGET=development
+
+WORKDIR /devel
+RUN chown -R scu:scu "${VIRTUAL_ENV}"
+
+
+HEALTHCHECK \
+  --interval=60s \
+  --timeout=60s \
+  --start-period=10s \
+  --retries=3 \
+  CMD ["echo", , "Fake healthcheck because it takes very long to boot in devel mode"]
```

Comment:
Is there no other way to remove it? This is a bit strage.

Reply:
The problem is that in devel-mode we change permissions and install packages on boot and that takes quite some time ... 
At the time the service really starts, it gets shutdown by the swarm because during a long startup it is reporting unhealthy.

IMO healthcheck does not make sense in devel mode. I added it here for completeness because in *some* of our services have it